internal/engine/upper_test.go

```diff
@@ -272,16 +283,41 @@ func (b *fakeBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.RSto
 		b.resultsByID[key] = val
 	}
 
+	// block until we know we're supposed to resolve this build
+	for {
+		if _, ok := b.buildsAwaitingResolution[count]; ok {
```

Comment:
fine as is, but if the values were chans instead of bools, this might be a bit cleaner (no loop needed). up to you.

Reply:
oooh neat! ty